### PR TITLE
Update DG to close issues

### LINKED ISSUE #212: [PE-D][Tester B] Class Diagram extra icon element
**Bug**:
The class diagram format has symbols like C and I which is not consistent which what is in the lecture slides:

**In DG**:

**In lecture slides**:

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: NCF3535/ped#9

### LINKED ISSUE #219: [PE-D][Tester B] Missing Glossary
**Bug**:
Missing glossary in the Developer Guide, instead, what was seen was `[ADD GLOSSARY HERE]`

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: NCF3535/ped#10

### PR DESCRIPTION
**Pull request by yakultbottle**
Closes #212, Closes #219

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -262,43 +320,52 @@ complexity.
...
+  form of monetary transaction.
 * Project Scope Constraints: Data storage is only to be performed locally.
-* Quality Requirements: The application should be able to be used effectively by a novice with little experience with CLIs.
+* Quality Requirements: The application should be able to be used effectively by a novice with little experience with
+  CLIs.
 
 ## Glossary
 
-[ADD GLOSSARY HERE]
+**Mainstream OS**: Windows, Linux, Unix, MacOS
```
Comment (yijiano) [inline docs/DeveloperGuide.md:334]: Perhaps this section has other terms you can consider adding in.

image

**Review by cxc0418 — APPROVED** [review]
Thank u for updating the DG

**Review by yijiano — APPROVED** [review]
LGTM, other than glossary

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -179,11 +181,61 @@ and analyze their inventory data through visual representation.
...
+    These exceptions are caught within the parseCommand method, allowing Parser
+    to handle error messages consistently across the application.
+
+#### Assumptions
+
+Parser expects commands to follow a specific format. For instance, commands
+like `add` require additional arguments, while commands like `list` and `exit`
+require none. Only recognized commands will proceed to execution; any unrecognized
+command results in a PillException.
+
```
Comment (cnivedit) [inline docs/DeveloperGuide.md:238]: Thanks for elaborating on the parser

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -262,43 +320,68 @@ complexity.
 
 ## Non-Functional Requirements
 
-* Technical Requirements: Any mainstream OS, i.e. Windows, macOS or Linux, with Java 11 installed. Instructions for downloading Java 11 can be found [here](https://www.oracle.com/sg/java/technologies/javase/jdk11-archive-downloads.html).
-* Project Scope Constraints: The application should only be used for tracking. It is not meant to be involved in any form of monetary transaction.
+* Technical Requirements: Any *mainstream OS* with Java 17 or above installed. 
+  Instructions for downloading Java 17 can be found 
+  [here](https://www.oracle.com/sg/java/technologies/javase/jdk17-archive-downloads.html).
+* Project Scope Constraints: The application should only be used for tracking. It is not meant to be involved in any
+  form of monetary transaction.
```
Comment (cnivedit) [inline docs/DeveloperGuide.md:327]: Good work fixing the formatting and spacing

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -262,43 +320,68 @@ complexity.
...
+    to the user (output).
+- **JUnit**: A testing framework for Java that allows developers to write and run
+    repeatable automated tests.
+- **Gradle**: A build automation tool for Java (and other languages) used to compile, test, and package applications.
+- **Text UI Testing**: A form of testing where interactions with a command-line
+    interface are tested by simulating user input and validating the application’s text output.
+- **Pharmaceutical Inventory**: Refers to the stock of medicines and other medical
+    supplies maintained by a pharmacy or healthcare facility for dispensing and logistics.
+- **Expiry Date**: The date after which a pharmaceutical product is no longer considered safe or effective for use.
+- **Restocking**: The process of replenishing inventory to ensure sufficient supply.
```
Comment (cnivedit) [inline docs/DeveloperGuide.md:350]: Thanks for adding in the glossary

File: docs/diagrams/Item-ClassDiagram.png
Comment (cnivedit) [inline docs/diagrams/Item-ClassDiagram.png:1]: Good work ensuring that the class diagram follows the conventions

**Review by cnivedit — APPROVED** [review]
LGTM. Thanks for updating the developer guide

**Pull request merged**